profiles/killbill/src/main/java/org/killbill/billing/server/filters/RequestDataFilter.java

```diff
@@ -58,33 +57,23 @@ public void filter(final ContainerRequestContext requestContext) {
 
         final List<String> kbEncodingHeaders = requestContext.getHeaders().get(KILL_BILL_ENCODING_HEADER);
         final String kbEncodingHeader = (kbEncodingHeaders == null || kbEncodingHeaders.isEmpty()) ? null : kbEncodingHeaders.get(0);
-        if (kbEncodingHeader != null) {
+        if ("base64".equalsIgnoreCase(kbEncodingHeader)) {
```

Comment:
https://github.com/killbill/killbill/pull/1573#discussion_r794404077 😁 

But I'm fine with this, since @sbrossie suggested it initially as well...